Fix Bootstrapping

tn3rb opened the pull request:
Garth was unable to activate AUEN on the Barista site because the following fatal error was occuring:

```
PHP Fatal error:  Uncaught RuntimeException:
Invalid arguments supplied for the EventEspresso\AutomatedUpcomingEventNotifications\domain\Domain class,
2 were required but 0 were passed. in /core/EE_Registry.core.php:1264

Stack trace:
#0 /core/EE_Registry.core.php(773): EE_Registry->_create_object()
#1 /core/services/loaders/CoreLoader.php(111): EE_Registry->create()
#2 /core/services/loaders/CachingLoader.php(147): EventEspresso\core\services\loaders\CoreLoader->load()
#3 /core/services/loaders/Loader.php(82): EventEspresso\core\services\loaders\CachingLoader->load()
#4 /core/EE_Registry.core.php(1436): EventEspresso\core\services\loaders\Loader->load()
#5 in /core/EE_Registry.core.php on line 1264
```

This PR:

- adds a dependency map loader function for the auen domain

- uses that for bootstrapping

manzoorwanijk reviewed (approved):
Looks great

Pebblo commented:
@tn3rb this branch is working fine on EDTR/dev but it's causing a memory leak on 4.10.XX

With only EE4 master active (almost new site, single event added) if I enable AUEN with this branch is get memory exhausted errors up to 512M and 1024M.

tn3rb commented:
kk thnx Tony, I'll look into this more

tn3rb commented:
I moved all of the bootstrap logic into a class in order to make some of the loading and depency registration shenanigans a little easier.
I also nuked what appeared to be an unnecessary module and moved its single line of code into the plugin's main addon class.

manzoorwanijk commented on domain/Bootstrap.php:
```diff
@@ -0,0 +1,136 @@
+            DomainInterface::class,
+            RegisterCustomShortcodeLibrary::class
+        );
+        $loader = function () {
+            return Bootstrap::getAuenDomain();
+        };
```
or
```suggestion
       $loader = [Bootstrap::class, 'getAuenDomain'];
```

tn3rb replied:
good suggestion, but the logic in `EE_Dependency_Map` and the loaders is specifically expecting a Closure and unfortunately won't work even if the supplied value is a valid callable.

manzoorwanijk replied:
Just out of curiosity, is there a way in PHP to differentiate between the two - closures vs callable?

tn3rb commented:
can you try this again plz Tony?

Pebblo commented:
Testing with the latest version of Master and the fix-bootstrapping branch of AUEN I'm getting a fatal:

```


<!--StartFragment-->
Fatal error: Uncaught Error: Call to undefined method EE_Dependency_Map::registerClassLoader() in C:\laragon\www\ee4editor\wp-content\plugins\eea-automated-upcoming-event-notifications\domain\Bootstrap.php on line&nbsp;65
--



1 | 0.0011 | 565400 | {main}( ) | ...\plugins.php:0
2 | 0.0016 | 614296 | require_once(&nbsp;'C:\laragon\www\ee4editor\wp-admin\admin.php&nbsp;) | ...\plugins.php:10
3 | 0.0018 | 629032 | require_once(&nbsp;'C:\laragon\www\ee4editor\wp-load.php&nbsp;) | ...\admin.php:34
4 | 0.0021 | 640240 | require_once(&nbsp;'C:\laragon\www\ee4editor\wp-config.php&nbsp;) | ...\wp-load.php:37
5 | 0.0028 | 736320 | require_once(&nbsp;'C:\laragon\www\ee4editor\wp-settings.php&nbsp;) | ...\wp-config.php:108
6 | 0.2053 | 25026000 | do_action(&nbsp;$tag =&nbsp;'plugins_loaded'&nbsp;) | ...\wp-settings.php:423
7 | 0.2053 | 25026376 | WP_Hook-&gt;do_action(&nbsp;$args =&nbsp;[0 =&gt; '']&nbsp;) | ...\plugin.php:484
8 | 0.2053 | 25026376 | WP_Hook-&gt;apply_filters(&nbsp;$value =&nbsp;'',&nbsp;$args =&nbsp;[0 =&gt; '']&nbsp;) | ...\class-wp-hook.php:316
9 | 0.2902 | 28291824 | EE_Bootstrap::load_espresso_addons(&nbsp;''&nbsp;) | ...\class-wp-hook.php:292
10 | 0.2903 | 28291824 | do_action(&nbsp;$tag =&nbsp;'AHEE__EE_Bootstrap__load_espresso_addons'&nbsp;) | ...\EE_Bootstrap.core.php:25
11 | 0.2903 | 28292200 | WP_Hook-&gt;do_action(&nbsp;$args =&nbsp;[0 =&gt; '']&nbsp;) | ...\plugin.php:484
12 | 0.2903 | 28292200 | WP_Hook-&gt;apply_filters(&nbsp;$value =&nbsp;'',&nbsp;$args =&nbsp;[0 =&gt; '']&nbsp;) | ...\class-wp-hook.php:316
13 | 0.2955 | 28358456 | EE_System-&gt;load_espresso_addons(&nbsp;''&nbsp;) | ...\class-wp-hook.php:292
14 | 0.2955 | 28358456 | do_action(&nbsp;$tag =&nbsp;'AHEE__EE_System__load_espresso_addons'&nbsp;) | ...\EE_System.core.php:383
15 | 0.2955 | 28358832 | WP_Hook-&gt;do_action(&nbsp;$args =&nbsp;[0 =&gt; '']&nbsp;) | ...\plugin.php:484
16 | 0.2955 | 28358832 | WP_Hook-&gt;apply_filters(&nbsp;$value =&nbsp;'',&nbsp;$args =&nbsp;[0 =&gt; '']&nbsp;) | ...\class-wp-hook.php:316
17 | 0.2955 | 28359960 | EventEspresso\AutomatedUpcomingEventNotifications\domain\Bootstrap-&gt;loadAutomatedUpcomingEventNotifications(&nbsp;''&nbsp;) | ...\class-wp-hook.php:292
18 | 0.2955 | 28359960 | EventEspresso\AutomatedUpcomingEventNotifications\domain\Bootstrap-&gt;registerDependencies( ) | ...\Bootstrap.php:44

<!--EndFragment-->

```

tn3rb commented:
oops... sry about that Tony

fixed now and confirmed activation with `4.10.12.rc.005` (`master` branch), as well as ` 4.11.0.rc.001` (`dev` branch)

Pebblo commented:
Thanks, Brent. This has fixed the issue here but created a new one.

As it's related to these changes I'll add the error here:

```
Fatal error: Uncaught TypeError: Argument 1 passed to EventEspresso\core\domain\services\validation\email\EmailValidationService::__construct() must be an instance of EE_Registration_Config, null given in C:\laragon\www\ee4editor\wp-content\plugins\event-espresso-core\core\domain\services\validation\email\EmailValidationService.php on line 37

TypeError: Argument 1 passed to EventEspresso\core\domain\services\validation\email\EmailValidationService::__construct() must be an instance of EE_Registration_Config, null given in C:\laragon\www\ee4editor\wp-content\plugins\event-espresso-core\core\domain\services\validation\email\EmailValidationService.php on line 37
```

This is thrown with the latest dev branch and happens when you edit any message template if the AUEN add-on has been activated.

Want another ticket for it?

Pebblo commented:
No. Only with AUEN.

Pebblo commented:
Here's the full stack:

```


<!--StartFragment-->
( ! )&nbsp;Fatal error: Uncaught TypeError: Argument 1 passed to EventEspresso\core\domain\services\validation\email\EmailValidationService::__construct() must be an instance of EE_Registration_Config, null given in C:\laragon\www\ee4editor\wp-content\plugins\event-espresso-core\core\domain\services\validation\email\EmailValidationService.php on line&nbsp;37
--



1 | 0.0006 | 463272 | {main}( ) | ...\admin.php:0
2 | 0.0008 | 478072 | require_once(&nbsp;'C:\laragon\www\ee4editor\wp-load.php&nbsp;) | ...\admin.php:34
3 | 0.0010 | 489360 | require_once(&nbsp;'C:\laragon\www\ee4editor\wp-config.php&nbsp;) | ...\wp-load.php:37
4 | 0.0016 | 585664 | require_once(&nbsp;'C:\laragon\www\ee4editor\wp-settings.php&nbsp;) | ...\wp-config.php:108
5 | 0.2177 | 24925784 | do_action(&nbsp;$tag =&nbsp;'plugins_loaded'&nbsp;) | ...\wp-settings.php:423
6 | 0.2177 | 24926160 | WP_Hook-&gt;do_action(&nbsp;$args =&nbsp;[0 =&gt; '']&nbsp;) | ...\plugin.php:484
7 | 0.2177 | 24926160 | WP_Hook-&gt;apply_filters(&nbsp;$value =&nbsp;'',&nbsp;$args =&nbsp;[0 =&gt; '']&nbsp;) | ...\class-wp-hook.php:316
8 | 0.3137 | 28179288 | EE_Bootstrap::load_espresso_addons(&nbsp;''&nbsp;) | ...\class-wp-hook.php:292
9 | 0.3137 | 28179288 | do_action(&nbsp;$tag =&nbsp;'AHEE__EE_Bootstrap__load_espresso_addons'&nbsp;) | ...\EE_Bootstrap.core.php:25
10 | 0.3137 | 28179664 | WP_Hook-&gt;do_action(&nbsp;$args =&nbsp;[0 =&gt; '']&nbsp;) | ...\plugin.php:484
11 | 0.3137 | 28179664 | WP_Hook-&gt;apply_filters(&nbsp;$value =&nbsp;'',&nbsp;$args =&nbsp;[0 =&gt; '']&nbsp;) | ...\class-wp-hook.php:316
12 | 0.3183 | 28245920 | EE_System-&gt;load_espresso_addons(&nbsp;''&nbsp;) | ...\class-wp-hook.php:292
13 | 0.3380 | 29189504 | do_action(&nbsp;$tag =&nbsp;'AHEE__EE_System__load_espresso_addons__complete'&nbsp;) | ...\EE_System.core.php:401
14 | 0.3380 | 29189880 | WP_Hook-&gt;do_action(&nbsp;$args =&nbsp;[0 =&gt; '']&nbsp;) | ...\plugin.php:484
15 | 0.3380 | 29189880 | WP_Hook-&gt;apply_filters(&nbsp;$value =&nbsp;'',&nbsp;$args =&nbsp;[0 =&gt; '']&nbsp;) | ...\class-wp-hook.php:316
16 | 0.3380 | 29194072 | EventEspresso\AutomatedUpcomingEventNotifications\domain\AutomatedUpcomingEventNotifications-&gt;after_registration(&nbsp;''&nbsp;) | ...\class-wp-hook.php:292
17 | 0.3382 | 29196704 | EventEspresso\core\services\loaders\Loader-&gt;getShared(&nbsp;$fqcn =&nbsp;'EventEspresso\\AutomatedUpcomingEventNotifications\\domain\\services\\admin\\Controller',&nbsp;$arguments =&nbsp;??? ) | ...\AutomatedUpcomingEventNotifications.php:129
18 | 0.3382 | 29196704 | EventEspresso\core\services\loaders\Loader-&gt;load(&nbsp;$fqcn =&nbsp;'EventEspresso\\AutomatedUpcomingEventNotifications\\domain\\services\\admin\\Controller',&nbsp;$arguments =&nbsp;[],&nbsp;$shared =&nbsp;??? ) | ...\Loader.php:105
19 | 0.3388 | 29228432 | EventEspresso\core\services\loaders\CachingLoader-&gt;load(&nbsp;$fqcn =&nbsp;'EventEspresso\\AutomatedUpcomingEventNotifications\\domain\\services\\admin\\Controller',&nbsp;$arguments =&nbsp;[],&nbsp;$shared =&nbsp;TRUE,&nbsp;$interfaces =&nbsp;??? ) | ...\Loader.php:82
20 | 0.3388 | 29228432 | EventEspresso\core\services\loaders\CoreLoader-&gt;load(&nbsp;$fqcn =&nbsp;'EventEspresso\\AutomatedUpcomingEventNotifications\\domain\\services\\admin\\Controller',&nbsp;$arguments =&nbsp;[],&nbsp;$shared =&nbsp;TRUE&nbsp;) | ...\CachingLoader.php:147
21 | 0.3388 | 29228488 | EE_Registry-&gt;create(&nbsp;$class_name =&nbsp;'EventEspresso\\AutomatedUpcomingEventNotifications\\domain\\services\\admin\\Controller',&nbsp;$arguments =&nbsp;[],&nbsp;$cache =&nbsp;TRUE,&nbsp;$from_db =&nbsp;FALSE,&nbsp;$load_only =&nbsp;FALSE,&nbsp;$addon =&nbsp;FALSE&nbsp;) | ...\CoreLoader.php:111
22 | 0.3390 | 29228600 | EE_Registry-&gt;_create_object(&nbsp;$class_name =&nbsp;'EventEspresso\\AutomatedUpcomingEventNotifications\\domain\\services\\admin\\Controller',&nbsp;$arguments =&nbsp;[],&nbsp;$type =&nbsp;'',&nbsp;$from_db =&nbsp;FALSE&nbsp;) | ...\EE_Registry.core.php:773
23 | 0.3391 | 29231576 | newInstanceArgs(&nbsp;$args =&nbsp;[0 =&gt; class EventEspresso\core\services\request\Request { private $get = [...]; private $post = [...]; private $cookie = [...]; private $server = [...]; private $files = [...]; private $request = [...]; private $request_type = class EventEspresso\core\domain\services\contexts\RequestTypeContextChecker { ... }; private $ip_address = '127.0.0.1'; private $user_agent = 'Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/90.0.4430.212 Safari/537.36 '; private $is_bot = FALSE }]&nbsp;) | ...\EE_Registry.core.php:1262
24 | 0.3391 | 29231720 | EventEspresso\AutomatedUpcomingEventNotifications\domain\services\admin\Controller-&gt;__construct(&nbsp;$request =&nbsp;class EventEspresso\core\services\request\Request { private $get = ['page' =&gt; 'espresso_messages', 'action' =&gt; 'edit_message_template', 'id' =&gt; '14', 'context' =&gt; 'purchaser', 'edit_message_template_nonce' =&gt; 'c8cfe06950', 'return' =&gt; 'global_mtps']; private $post = []; private $cookie = ['wordpress_85a790b9c8c0d6519e77e31bcc6a665a' =&gt; 'P3bblo\|1622751374\|anRiMnrdh3KRatsW8v0haWlJfEW7Nd1KGAnGxGLU8j7\|c52b9846bd66c883d12dbeb2c4e2c0487160f055528217394e6ed2487972a680', 'wp-settings-1' =&gt; 'hidetb=1', 'wp-settings-time-1' =&gt; '1620829734', 'PHPSESSID' =&gt; 'dpdr83tqctlbtvhmbhfu3v54nt', 'wordpress_test_cookie' =&gt; 'WP Cookie check', 'wordpress_logged_in_85a790b9c8c0d6519e77e31bcc6a665a' =&gt; 'P3bblo\|1622751374\|anRiMnrdh3KRatsW8v0haWlJfEW7Nd1KGAnGxGLU8j7\|93f918f5a8e505a8b4a601cb9762f233a7271f151c6c1463c78f2f1d3ab64eeb']; private $server = ['SERVER_SOFTWARE' =&gt; 'nginx/1.14.0', 'REQUEST_URI' =&gt; '/wp-admin/admin.php?page=espresso_messages&amp;action=edit_message_template&amp;id=14&amp;context=purchaser&amp;edit_message_template_nonce=c8cfe06950&amp;return=global_mtps', 'ALLUSERSPROFILE' =&gt; 'C:\\ProgramData', 'APPDATA' =&gt; 'C:\\Users\\Tony\\AppData\\Roaming', 'ChocolateyInstall' =&gt; 'C:\\ProgramData\\chocolatey', 'ChocolateyLastPathUpdate' =&gt; '132342836895363948', 'CommonProgramFiles' =&gt; 'C:\\Program Files\\Common Files', 'CommonProgramFiles(x86)' =&gt; 'C:\\Program Files (x86)\\Common Files', 'CommonProgramW6432' =&gt; 'C:\\Program Files\\Common Files', 'COMPUTERNAME' =&gt; 'TONY-PC', 'ComSpec' =&gt; 'C:\\WINDOWS\\system32\\cmd.exe', 'DriverData' =&gt; 'C:\\Windows\\System32\\Drivers\\DriverData', 'FPS_BROWSER_APP_PROFILE_STRING' =&gt; 'Internet Explorer', 'FPS_BROWSER_USER_PROFILE_STRING' =&gt; 'Default', 'HOMEDRIVE' =&gt; 'C:', 'HOMEPATH' =&gt; '\\Users\\Tony', 'LOCALAPPDATA' =&gt; 'C:\\Users\\Tony\\AppData\\Local', 'LOGONSERVER' =&gt; '\\\\TONY-PC', 'NUMBER_OF_PROCESSORS' =&gt; '4', 'OneDrive' =&gt; 'C:\\Users\\Tony\\OneDrive', 'OneDriveConsumer' =&gt; 'C:\\Users\\Tony\\OneDrive', 'OS' =&gt; 'Windows_NT', 'Path' =&gt; 'C:\\Program Files (x86)\\NVIDIA Corporation\\PhysX\\Common;C:\\Program Files (x86)\\Common Files\\Oracle\\Java\\javapath;C:\\Python38\\Scripts\\;C:\\Python38\\;C:\\WINDOWS\\system32;C:\\WINDOWS;C:\\WINDOWS\\System32\\Wbem;C:\\WINDOWS\\System32\\WindowsPowerShell\\v1.0\\;C:\\WINDOWS\\System32\\OpenSSH\\;C:\\Program Files\\PuTTY\\;C:\\Program Files\\nodejs\\;C:\\ProgramData\\chocolatey\\bin;C:\\Program Files\\Git\\cmd;C:\\laragon\\bin\\php\\php-7.3.7-Win32-VC15-x64;C:\\ProgramData\\ComposerSetup\\bin;C:\\Prog', 'PATHEXT' =&gt; '.COM;.EXE;.BAT;.CMD;.VBS;.VBE;.JS;.JSE;.WSF;.WSH;.MSC;.PY;.PYW', 'PhpStorm' =&gt; 'C:\\Program Files\\JetBrains\\PhpStorm 2020.1.4\\bin;', 'PROCESSOR_ARCHITECTURE' =&gt; 'AMD64', 'PROCESSOR_IDENTIFIER' =&gt; 'Intel64 Family 6 Model 60 Stepping 3, GenuineIntel', 'PROCESSOR_LEVEL' =&gt; '6', 'PROCESSOR_REVISION' =&gt; '3c03', 'ProgramData' =&gt; 'C:\\ProgramData', 'ProgramFiles' =&gt; 'C:\\Program Files', 'ProgramFiles(x86)' =&gt; 'C:\\Program Files (x86)', 'ProgramW6432' =&gt; 'C:\\Program Files', 'PSModulePath' =&gt; 'C:\\Program Files\\WindowsPowerShell\\Modules;C:\\WINDOWS\\system32\\WindowsPowerShell\\v1.0\\Modules', 'PUBLIC' =&gt; 'C:\\Users\\Public', 'SESSIONNAME' =&gt; 'Console', 'SystemDrive' =&gt; 'C:', 'SystemRoot' =&gt; 'C:\\WINDOWS', 'TEMP' =&gt; 'C:\\Users\\Tony\\AppData\\Local\\Temp', 'TMP' =&gt; 'C:\\Users\\Tony\\AppData\\Local\\Temp', 'USERDOMAIN' =&gt; 'TONY-PC', 'USERDOMAIN_ROAMINGPROFILE' =&gt; 'TONY-PC', 'USERNAME' =&gt; 'Tony', 'USERPROFILE' =&gt; 'C:\\Users\\Tony', 'windir' =&gt; 'C:\\WINDOWS', 'HTTP_COOKIE' =&gt; 'wordpress_85a790b9c8c0d6519e77e31bcc6a665a=P3bblo%7C1622751374%7CanRiMnrdh3KRatsW8v0haWlJfEW7Nd1KGAnGxGLU8j7%7Cc52b9846bd66c883d12dbeb2c4e2c0487160f055528217394e6ed2487972a680; wp-settings-1=hidetb%3D1; wp-settings-time-1=1620829734; PHPSESSID=dpdr83tqctlbtvhmbhfu3v54nt; wordpress_test_cookie=WP+Cookie+check; wordpress_logged_in_85a790b9c8c0d6519e77e31bcc6a665a=P3bblo%7C1622751374%7CanRiMnrdh3KRatsW8v0haWlJfEW7Nd1KGAnGxGLU8j7%7C93f918f5a8e505a8b4a601cb9762f233a7271f151c6c1463c78f2f1d3ab64eeb', 'HTTP_ACCEPT_LANGUAGE' =&gt; 'en-GB,en;q=0.9,en-US;q=0.8', 'HTTP_ACCEPT_ENCODING' =&gt; 'gzip, deflate', 'HTTP_REFERER' =&gt; 'http://ee4editor.test/wp-admin/admin.php?page=espresso_messages&amp;action=global_mtps&amp;global_mtps_nonce=f49021efbf', 'HTTP_ACCEPT' =&gt; 'text/html,application/xhtml+xml,application/xml;q=0.9,image/avif,image/webp,image/apng,*/*;q=0.8,application/signed-exchange;v=b3;q=0.9', 'HTTP_USER_AGENT' =&gt; 'Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/90.0.4430.212 Safari/537.36', 'HTTP_UPGRADE_INSECURE_REQUESTS' =&gt; '1', 'HTTP_CACHE_CONTROL' =&gt; 'max-age=0', 'HTTP_CONNECTION' =&gt; 'keep-alive', 'HTTP_HOST' =&gt; 'ee4editor.test', 'REDIRECT_STATUS' =&gt; '200', 'SERVER_NAME' =&gt; 'ee4editor.test', 'SERVER_PORT' =&gt; '80', 'SERVER_ADDR' =&gt; '127.0.0.1', 'REMOTE_PORT' =&gt; '64273', 'REMOTE_ADDR' =&gt; '127.0.0.1', 'GATEWAY_INTERFACE' =&gt; 'CGI/1.1', 'REQUEST_SCHEME' =&gt; 'http', 'SERVER_PROTOCOL' =&gt; 'HTTP/1.1', 'DOCUMENT_ROOT' =&gt; 'C:/laragon/www/ee4editor', 'DOCUMENT_URI' =&gt; '/wp-admin/admin.php', 'SCRIPT_NAME' =&gt; '/wp-admin/admin.php', 'CONTENT_LENGTH' =&gt; '', 'CONTENT_TYPE' =&gt; '', 'REQUEST_METHOD' =&gt; 'GET', 'QUERY_STRING' =&gt; 'page=espresso_messages&amp;action=edit_message_template&amp;id=14&amp;context=purchaser&amp;edit_message_template_nonce=c8cfe06950&amp;return=global_mtps', 'SCRIPT_FILENAME' =&gt; 'C:/laragon/www/ee4editor/wp-admin/admin.php', 'PATH_INFO' =&gt; '', 'FCGI_ROLE' =&gt; 'RESPONDER', 'PHP_SELF' =&gt; '/wp-admin/admin.php', 'REQUEST_TIME_FLOAT' =&gt; 1622585550.6814, 'REQUEST_TIME' =&gt; 1622585550]; private $files = []; private $request = ['page' =&gt; 'espresso_messages', 'action' =&gt; 'edit_message_template', 'id' =&gt; '14', 'context' =&gt; 'purchaser', 'edit_message_template_nonce' =&gt; 'c8cfe06950', 'return' =&gt; 'global_mtps']; private $request_type = class EventEspresso\core\domain\services\contexts\RequestTypeContextChecker { private $request_type = class EventEspresso\core\domain\entities\contexts\RequestTypeContext { ... }; private ${EventEspresso\core\services\context\ContextChecker}identifier = 'requesttypecontextchecker'; private ${EventEspresso\core\services\context\ContextChecker}acceptable_values = [...]; private ${EventEspresso\core\services\context\ContextChecker}evaluation_callback = class Closure { ... } }; private $ip_address = '127.0.0.1'; private $user_agent = 'Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/90.0.4430.212 Safari/537.36 '; private $is_bot = FALSE }&nbsp;) | ...\EE_Registry.core.php:1262
25 | 0.3391 | 29231720 | EventEspresso\AutomatedUpcomingEventNotifications\domain\services\admin\Controller-&gt;canLoad( ) | ...\Controller.php:65
26 | 0.3391 | 29231720 | EventEspresso\AutomatedUpcomingEventNotifications\domain\services\admin\Controller-&gt;messageTemplateGroup(&nbsp;$id =&nbsp;??? ) | ...\Controller.php:124
27 | 0.3520 | 30445744 | EEM_Message_Template_Group-&gt;get_one_by_ID(&nbsp;$id =&nbsp;14&nbsp;) | ...\Controller.php:207
28 | 0.3520 | 30447192 | EEM_Message_Template_Group-&gt;get_one(&nbsp;$query_params =&nbsp;['default_where_conditions' =&gt; 'minimum', 0 =&gt; ['GRP_ID' =&gt; 14]]&nbsp;) | ...\EEM_Base.model.php:1132
29 | 0.3520 | 30447568 | EEM_Message_Template_Group-&gt;get_all(&nbsp;$query_params =&nbsp;['default_where_conditions' =&gt; 'minimum', 0 =&gt; ['GRP_ID' =&gt; 14], 'limit' =&gt; 1]&nbsp;) | ...\EEM_Base.model.php:1195
30 | 0.3543 | 30471568 | EEM_Message_Template_Group-&gt;_create_objects(&nbsp;$rows =&nbsp;[0 =&gt; ['Message_Template_Group.GRP_ID' =&gt; '14', 'Message_Template_Group.MTP_name' =&gt; '', 'Message_Template_Group.MTP_description' =&gt; '', 'Message_Template_Group.MTP_user_id' =&gt; '1', 'Message_Template_Group.MTP_messenger' =&gt; 'html', 'Message_Template_Group.MTP_message_type' =&gt; 'invoice', 'Message_Template_Group.MTP_is_global' =&gt; '1', 'Message_Template_Group.MTP_is_override' =&gt; '0', 'Message_Template_Group.MTP_deleted' =&gt; '0', 'Message_Template_Group.MTP_is_active' =&gt; '1']]&nbsp;) | ...\EEM_Base.model.php:876
31 | 0.3668 | 31358040 | EEM_WP_User-&gt;instantiate_class_from_array_or_object(&nbsp;$cols_n_values =&nbsp;['Message_Template_Group.GRP_ID' =&gt; '14', 'Message_Template_Group.MTP_name' =&gt; '', 'Message_Template_Group.MTP_description' =&gt; '', 'Message_Template_Group.MTP_user_id' =&gt; '1', 'Message_Template_Group.MTP_messenger' =&gt; 'html', 'Message_Template_Group.MTP_message_type' =&gt; 'invoice', 'Message_Template_Group.MTP_is_global' =&gt; '1', 'Message_Template_Group.MTP_is_override' =&gt; '0', 'Message_Template_Group.MTP_deleted' =&gt; '0', 'Message_Template_Group.MTP_is_active' =&gt; '1']&nbsp;) | ...\EEM_Base.model.php:5141
32 | 0.3668 | 31358040 | EEM_WP_User-&gt;_deduce_fields_n_values_from_cols_n_values(&nbsp;$cols_n_values =&nbsp;['Message_Template_Group.GRP_ID' =&gt; '14', 'Message_Template_Group.MTP_name' =&gt; '', 'Message_Template_Group.MTP_description' =&gt; '', 'Message_Template_Group.MTP_user_id' =&gt; '1', 'Message_Template_Group.MTP_messenger' =&gt; 'html', 'Message_Template_Group.MTP_message_type' =&gt; 'invoice', 'Message_Template_Group.MTP_is_global' =&gt; '1', 'Message_Template_Group.MTP_is_override' =&gt; '0', 'Message_Template_Group.MTP_deleted' =&gt; '0', 'Message_Template_Group.MTP_is_active' =&gt; '1']&nbsp;) | ...\EEM_Base.model.php:5242
33 | 0.3668 | 31358512 | EE_Email_Field-&gt;prepare_for_set(&nbsp;$email_address =&nbsp;NULL&nbsp;) | ...\EEM_Base.model.php:5403
34 | 0.3673 | 31362440 | EventEspresso\core\domain\services\factories\EmailAddressFactory::create(&nbsp;$email_address =&nbsp;NULL&nbsp;) | ...\EE_Email_Field.php:46
35 | 0.3673 | 31362816 | EventEspresso\core\services\loaders\Loader-&gt;getNew(&nbsp;$fqcn =&nbsp;'EventEspresso\\core\\domain\\values\\EmailAddress',&nbsp;$arguments =&nbsp;[0 =&gt; NULL]&nbsp;) | ...\EmailAddressFactory.php:34
36 | 0.3673 | 31362816 | EventEspresso\core\services\loaders\Loader-&gt;load(&nbsp;$fqcn =&nbsp;'EventEspresso\\core\\domain\\values\\EmailAddress',&nbsp;$arguments =&nbsp;[0 =&gt; NULL],&nbsp;$shared =&nbsp;FALSE&nbsp;) | ...\Loader.php:94
37 | 0.3675 | 31367448 | EventEspresso\core\services\loaders\CoreLoader-&gt;load(&nbsp;$fqcn =&nbsp;'EventEspresso\\core\\domain\\values\\EmailAddress',&nbsp;$arguments =&nbsp;[0 =&gt; NULL],&nbsp;$shared =&nbsp;FALSE&nbsp;) | ...\Loader.php:83
38 | 0.3675 | 31367824 | EE_Registry-&gt;create(&nbsp;$class_name =&nbsp;'EventEspresso\\core\\domain\\values\\EmailAddress',&nbsp;$arguments =&nbsp;[0 =&gt; NULL],&nbsp;$cache =&nbsp;FALSE,&nbsp;$from_db =&nbsp;FALSE,&nbsp;$load_only =&nbsp;FALSE,&nbsp;$addon =&nbsp;FALSE&nbsp;) | ...\CoreLoader.php:111
39 | 0.3675 | 31367824 | EE_Registry-&gt;_create_object(&nbsp;$class_name =&nbsp;'EventEspresso\\core\\domain\\values\\EmailAddress',&nbsp;$arguments =&nbsp;[0 =&gt; NULL],&nbsp;$type =&nbsp;'',&nbsp;$from_db =&nbsp;FALSE&nbsp;) | ...\EE_Registry.core.php:773
40 | 0.3676 | 31367936 | EE_Registry-&gt;_resolve_dependencies(&nbsp;$reflector =&nbsp;class ReflectionClass { public $name = 'EventEspresso\\core\\domain\\values\\EmailAddress' },&nbsp;$class_name =&nbsp;'EventEspresso\\core\\domain\\values\\EmailAddress',&nbsp;$arguments =&nbsp;[0 =&gt; NULL]&nbsp;) | ...\EE_Registry.core.php:1231
41 | 0.3676 | 31370592 | EE_Registry-&gt;_resolve_dependency(&nbsp;$class_name =&nbsp;'EventEspresso\\core\\domain\\values\\EmailAddress',&nbsp;$param_class =&nbsp;'EventEspresso\\core\\domain\\services\\validation\\email\\EmailValidationService',&nbsp;$arguments =&nbsp;[0 =&gt; NULL],&nbsp;$index =&nbsp;1&nbsp;) | ...\EE_Registry.core.php:1362
42 | 0.3677 | 31370704 | EventEspresso\core\services\loaders\Loader-&gt;load(&nbsp;$fqcn =&nbsp;'EventEspresso\\core\\domain\\services\\validation\\email\\EmailValidationService',&nbsp;$arguments =&nbsp;[],&nbsp;$shared =&nbsp;TRUE&nbsp;) | ...\EE_Registry.core.php:1421
43 | 0.3677 | 31371080 | EventEspresso\core\services\loaders\CachingLoader-&gt;load(&nbsp;$fqcn =&nbsp;'EventEspresso\\core\\domain\\services\\validation\\email\\EmailValidationService',&nbsp;$arguments =&nbsp;[],&nbsp;$shared =&nbsp;TRUE,&nbsp;$interfaces =&nbsp;??? ) | ...\Loader.php:82
44 | 0.3678 | 31371080 | EventEspresso\core\services\loaders\CoreLoader-&gt;load(&nbsp;$fqcn =&nbsp;'EventEspresso\\core\\domain\\services\\validation\\email\\EmailValidationService',&nbsp;$arguments =&nbsp;[],&nbsp;$shared =&nbsp;TRUE&nbsp;) | ...\CachingLoader.php:147
45 | 0.3678 | 31371136 | EE_Registry-&gt;create(&nbsp;$class_name =&nbsp;'EventEspresso\\core\\domain\\services\\validation\\email\\EmailValidationService',&nbsp;$arguments =&nbsp;[],&nbsp;$cache =&nbsp;TRUE,&nbsp;$from_db =&nbsp;FALSE,&nbsp;$load_only =&nbsp;FALSE,&nbsp;$addon =&nbsp;FALSE&nbsp;) | ...\CoreLoader.php:111
46 | 0.3679 | 31371136 | EE_Registry-&gt;_create_object(&nbsp;$class_name =&nbsp;'EventEspresso\\core\\domain\\services\\validation\\email\\EmailValidationService',&nbsp;$arguments =&nbsp;[],&nbsp;$type =&nbsp;'',&nbsp;$from_db =&nbsp;FALSE&nbsp;) | ...\EE_Registry.core.php:773
47 | 0.3686 | 31375808 | newInstanceArgs(&nbsp;$args =&nbsp;[0 =&gt; NULL, 1 =&gt; class EventEspresso\core\services\loaders\Loader { private $new_loader = class EventEspresso\core\services\loaders\CoreLoader { ... }; private $shared_loader = class EventEspresso\core\services\loaders\CachingLoader { ... }; private $class_cache = class EventEspresso\core\services\loaders\ClassInterfaceCache { ... } }]&nbsp;) | ...\EE_Registry.core.php:1262
48 | 0.3686 | 31375952 | EventEspresso\core\domain\services\validation\email\EmailValidationService-&gt;__construct(&nbsp;$config =&nbsp;NULL,&nbsp;$loader =&nbsp;class EventEspresso\core\services\loaders\Loader { private $new_loader = class EventEspresso\core\services\loaders\CoreLoader { private $generator = class EE_Registry { ... } }; private $shared_loader = class EventEspresso\core\services\loaders\CachingLoader { protected $identifier = ''; protected $cache = class EventEspresso\core\services\collections\LooseCollection { ... }; private $object_identifier = class EventEspresso\core\services\loaders\ObjectIdentifier { ... }; protected $loader = class EventEspresso\core\services\loaders\CoreLoader { ... } }; private $class_cache = class EventEspresso\core\services\loaders\ClassInterfaceCache { private $interfaces = [...]; protected $aliases = [...] } }&nbsp;)

<!--EndFragment-->

```

tn3rb commented:
thnx for the stack, looking into that now

tn3rb commented:
try that now plz Tony

Pebblo commented:
@tn3rb I get the same error with master

Pebblo commented:
Yeah, looks like it's working now.

Thanks Brent.
